components/reusable-components/blogDetails-card/BlogDetailsCard.tsx

```diff
@@ -0,0 +1,305 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+
+import React, { useEffect, useState, useMemo } from 'react';
+import Image from 'next/image';
+import { useRouter, useSearchParams } from 'next/navigation';
+import { useDropzone } from 'react-dropzone';
+import DOMPurify from 'dompurify';
+import styles from '../../../app/content-panel/blogs/[id]/BlogDetailsPage.module.scss';
+import TiptapEditor from '../../editor/TiptapEditor';
+import 'bootstrap-icons/font/bootstrap-icons.css';
+
+interface BlogDetailsCardProps {
+  title: string;
+  imageUrl: string;
+  content: string;
+  author: { first_name: string; last_name: string };
+  publishDate: string;
+  tags: string[];
+  onImageChange: (files: File[]) => void;
+  onChange: (
+    event: React.ChangeEvent<HTMLInputElement> | { target: { name: string; value: string } }
+  ) => void;
+  onDeleteClick: () => void;
+  onCancelClick: () => void;
+  imageError: string | null;
+}
+
+const BlogDetailsCard: React.FC<BlogDetailsCardProps> = ({
+  title,
+  imageUrl,
+  content,
+  author: propAuthor,
+  publishDate,
+  tags,
+  onImageChange,
+  onChange,
+  onDeleteClick,
+  onCancelClick,
+}) => {
+  // TODO: The Author should be a dropdown where all the content managers would be listed and we can select one.
+  // There are 3 types of users which are: users, admins, and content managers, so according to role filter them and take only users that are content managers
+
+  const hardcodedAuthors = useMemo(
+    () => [
+      { id: '1', first_name: 'John', last_name: 'Doe' },
+      { id: '2', first_name: 'Jane', last_name: 'Smith' },
+      { id: '3', first_name: 'Emily', last_name: 'Johnson' },
+    ],
+    []
+  );
+
+  const [isEditable, setIsEditable] = useState<boolean>(false);
+  const [selectedAuthorId, setSelectedAuthorId] = useState<string>('');
+  const [imageFile, setImageFile] = useState<File | null>(null);
+  const [validationErrors, setValidationErrors] = useState<{ image: string | null }>({
+    image: null,
+  });
+
+  const {
+    getRootProps,
+    getInputProps,
+    acceptedFiles,
+    fileRejections: rejectedFiles,
+  } = useDropzone({
+    accept: {
+      'image/*': [],
+    },
+    maxSize: 5 * 1024 * 1024,
+    onDrop: (files) => {
+      if (files.length > 0) {
+        setImageFile(files[0]);
+        onImageChange(files);
+      }
+    },
+    onDropRejected: (rejectedFiles) => {
+      if (process.env.NODE_ENV !== 'production') {
+        console.log('File Rejections:', rejectedFiles);
+      }
+    },
+  });
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    const editMode = searchParams.get('edit') === 'true';
+    setIsEditable(editMode);
+  }, [searchParams]);
```

Comment:
I can see that you are adding edit=true in the searchParams to signal that the blog post is in edit mode. You can do that by just setting the setIsEditable on click of the Edit button. There is no need for this logic here.

Reply:
Initially, I implemented it the way you suggested, setting setIsEditable on the Edit button click. However, I ran into an issue when navigating from the BlogListViewPage. If I select "Edit" from the dropdown there, the inputs on the BlogDetailsPage remain non-editable until I manually click the Edit button again. That's why I added the edit=true logic in the searchParams to ensure the edit mode is enabled right away when the page loads.

Reply:
Should I change this ?

Reply:
You will run into the same issue, if you want to see the changes you have made while editing the blog post, when you return to the blog posts page.
How can both components share state? and listen/be aware of changes made?

Reply:
I used react-redux, is it ok ?

Reply:
Use Context for this. We just need some values to be available to both components.